buildchain/buildchain/versions.py

```diff
@@ -291,201 +294,113 @@ def deb_full_name(self) -> str:
+        PackageVersion(name='runc'),
+        PackageVersion(name='salt-minion', version=SALT_VERSION),
+        PackageVersion(name='socat'),
+        PackageVersion(name='sos'),  # TODO download built package dependencies
+        PackageVersion(name='util-linux'),
+        PackageVersion(name='xfsprogs'),
+    ),
+    'redhat': (
+        PackageVersion(
+            name='calico-cni-plugin',
+            version=CALICO_VERSION,
+            release='1.el7'
+        ),
+        PackageVersion(name='container-selinux'),  # TODO #1710
+        PackageVersion(
+            name='metalk8s-sosreport',
+            version=SHORT_VERSION,
+            release='1.el7'
+        ),
+        PackageVersion(name='yum-plugin-versionlock'),
     ),
-    PackageVersion(
-        name='calico-cni-plugin',
-        version=CALICO_VERSION,
-        release='1.el7',
+    'debian': (
+        PackageVersion(
+            name='calico-cni-plugin',
+            version=CALICO_VERSION,
+            release='0'
+        ),
+        PackageVersion(name='iproute2', override='iproute'),
+        PackageVersion(
+            name='metalk8s-sosreport',
+            version=SHORT_VERSION,
+            release='0'
+        ),
+        PackageVersion(name='python-m2crypto', override='m2crypto'),
+        PackageVersion(name='sosreport', override='sos'),
     ),
-)
+}
 
-PACKAGES_MAP = {pkg.name: pkg for pkg in RPM_PACKAGES}
 
+def _list_pkgs_for_os_family(os_family: str) -> Tuple[PackageVersion, ...]:
+    """List downloaded packages for a given OS family.
 
-DEB_PACKAGES = (
-    PackageVersion(
-        name='containerd',
-    ),
-    PackageVersion(
-        name='cri-tools',
-    ),
-    PackageVersion(
-        name='coreutils',
-    ),
-    PackageVersion(
-        name='ebtables',
-    ),
-    PackageVersion(
-        name='ethtool',
-    ),
-    PackageVersion(
-        name='e2fsprogs',
-    ),
-    PackageVersion(
-        name='genisoimage',
-    ),
-    PackageVersion(
-        name='iproute2',
-    ),
-    PackageVersion(
-        name='iptables',
-    ),
-    PackageVersion(
-        name='salt-minion',
-        version=SALT_VERSION,
-    ),
-    PackageVersion(
-        name='kubectl',
-        version=K8S_VERSION,
-    ),
-    PackageVersion(
-        name='kubelet',
-        version=K8S_VERSION,
-    ),
-    PackageVersion(
-        name='kubernetes-cni',
-    ),
-    PackageVersion(
-        name='python-m2crypto',
-    ),
-    PackageVersion(
-        name='runc',
-    ),
-    PackageVersion(
-        name='socat',
-    ),
-    PackageVersion(
-        name='sosreport',
-    ),
-    PackageVersion(
-        name='util-linux',
-    ),
-    PackageVersion(
-        name='xfsprogs',
-    ),
-)
+    Arguments:
+        os_family: OS_family for which to list packages
+    """
+    common_pkgs = PACKAGES['common']
+    os_family_pkgs = PACKAGES.get(os_family)
+
+    if os_family_pkgs is None:
+        raise Exception('Unlisted OS family: {}'.format(os_family))
```

Comment:
s/Unlisted/Unsupported/?

Reply:
Eeeeh technically we don't support Ubuntu yet but we download its packages. As `versions` is supposed to be absolutely standalone I'd rather observe things that pertain to it and only to it, hence unlisted in the context of the package dict keys.

Reply:
Make sense, but I think we can have a better word than `unlisted`.
Maybe `no packages for OS family {}`?

Reply:
Suits me, thanks